tests/integration/api.rs

```diff
@@ -28,127 +30,236 @@ use crate::mocks::mock_wallet::MockWallet;
+        send(unauthorized_request(
+            &settings,
+            Method::GET,
+            routes::LIST_CHANNELS
+        ))
+        .await
+        .unwrap_err()
+    );
+    assert_eq!(
+        StatusCode::UNAUTHORIZED,
+        send(unauthorized_request(
+            &settings,
+            Method::POST,
+            routes::OPEN_CHANNEL
+        ))
+        .await
+        .unwrap_err()
+    );
+    assert_eq!(
+        StatusCode::UNAUTHORIZED,
+        send(readonly_request_with_body(
+            &settings,
+            Method::POST,
+            routes::OPEN_CHANNEL,
+            fund_channel_request
+        )?)
+        .await
+        .unwrap_err()
+    );
+    assert_eq!(
+        StatusCode::UNAUTHORIZED,
+        send(unauthorized_request(
+            &settings,
+            Method::POST,
+            routes::WITHDRAW
+        ))
+        .await
+        .unwrap_err()
+    );
+    assert_eq!(
+        StatusCode::UNAUTHORIZED,
+        send(readonly_request_with_body(
+            &settings,
+            Method::POST,
+            routes::WITHDRAW,
+            withdraw_request
+        )?)
+        .await
+        .unwrap_err()
+    );
+    assert_eq!(
+        StatusCode::UNAUTHORIZED,
+        send(readonly_request_with_body(
+            &settings,
+            Method::GET,
+            routes::NEW_ADDR,
+            NewAddress::default
+        )?)
+        .await
+        .unwrap_err()
+    );
+    assert_eq!(
+        StatusCode::UNAUTHORIZED,
+        send(readonly_request_with_body(
+            &settings,
+            Method::GET,
+            routes::NEW_ADDR,
+            NewAddress::default
+        )?)
+        .await
+        .unwrap_err()
+    );
+    assert_eq!(
+        StatusCode::UNAUTHORIZED,
+        send(unauthorized_request(
+            &settings,
+            Method::GET,
+            routes::LIST_PEERS
+        ))
+        .await
+        .unwrap_err()
+    );
+    assert_eq!(
+        StatusCode::UNAUTHORIZED,
+        send(unauthorized_request(
+            &settings,
+            Method::POST,
+            routes::CONNECT_PEER
+        ))
+        .await
+        .unwrap_err()
+    );
+    assert_eq!(
+        StatusCode::UNAUTHORIZED,
+        send(readonly_request_with_body(
+            &settings,
+            Method::POST,
+            routes::CONNECT_PEER,
+            || TEST_ADDRESS
+        )?)
+        .await
+        .unwrap_err()
+    );
+    assert_eq!(
+        StatusCode::UNAUTHORIZED,
+        send(unauthorized_request(
+            &settings,
+            Method::DELETE,
+            routes::DISCONNECT_PEER
+        ))
+        .await
+        .unwrap_err()
+    );
+    assert_eq!(
+        StatusCode::UNAUTHORIZED,
+        send(readonly_request_with_body(
+            &settings,
+            Method::DELETE,
+            routes::DISCONNECT_PEER,
+            || TEST_ADDRESS
+        )?)
+        .await
+        .unwrap_err()
+    );
+    Ok(())
 }
 
-unauthorized!(
```

Comment:
I dropped the macro because it became quite hard to support with the explicit settings object.

Reply:
yeah I was thinking it better to remove it after I optimised it so much. I think its still better to have a test per API method though, its only small and makes it easier to see exactly what failed.

Reply:
Would you not get a line number for the assert that failed?

Reply:
Yes but its quicker to just read the name of the failed test (with API call in the name) instead of having to go to the line number in the test to find it. And we get more information about the problem if many or a single test fails, which we don't get if all the asserts are in one test.
And you have 128 threads to use :)

Reply:
I actually suspect that having separate tests would be slower than just doing simple requests serial since it also needs some test setup. I would leave it the refactoring to you since I don't know how to make it work without having to write a lot of complex macros or boilercode.